### PR TITLE
raft : write compact if statements

### PR DESCRIPTION
**Pull request by nilsocket**
write compact if statements, rather than making code lengthy,
functionality is not changed.

Thank you.

### CODE REVIEW

File: raft/log_unstable.go [hunk 1]
```diff
@@ -55,22 +55,16 @@ func (u *unstable) maybeLastIndex() (uint64, bool) {
...
-			return 0, false
-		}
-		if u.snapshot.Metadata.Index == i {
+		if u.snapshot != nil && u.snapshot.Metadata.Index == i {
 			return u.snapshot.Metadata.Term, true
 		}
 		return 0, false
 	}
 
-	last, ok := u.maybeLastIndex()
```
Comment (xiang90) [inline raft/log_unstable.go]: can we revert this change? i think the old one is more readable.

**Comment by xiang90** [comment]
I like the first change. but i think the second one is not needed.

File: raft/log_unstable.go (hunk 1 above)
Comment (nilsocket) [reply raft/log_unstable.go]: But, I think it's really a simple case, it's also very common to write such statements in go.

Anyhow, if you want it to be changed (just mention), I will change it.

Thank you.

Comment (xiang90) [reply raft/log_unstable.go]: please revert this change. thanks!

Comment (nilsocket) [reply raft/log_unstable.go]: made necessary changes

Thank you.

**Comment by xiang90** [comment]
@nilsocket Thanks! LGTM. Please squash the two commits into one. Then it will be ready to merge.

**Comment by nilsocket** [comment]
Sorry for the delay. squashed commits. Thank you.

**Comment by xiang90** [comment]
lgtm. thanks
